examples/notebooks/blog/unittests_in_beam.ipynb

```diff
@@ -357,6 +248,123 @@
       },
       "execution_count": null,
       "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Example 3**\n",
+        "\n",
+        "This `DoFn` (and corresponding pipeline) is used to convey a situation in which a `DoFn` makes an API call. Note that an error is raised here if the length of the API response (returned_record) is less than length 10."
+      ],
+      "metadata": {
+        "id": "Z8__izORM3r8"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Fake client to simulate an external call\n",
+        "\n",
+        "import time\n",
+        "class Client():\n",
+        "   def get_data(self, api):\n",
+        "      time.sleep(3)\n",
+        "      return [0,1,2,3,4,5,6,7,8,9]\n",
+        "\n",
+        "MyApiCall = Client()"
+      ],
+      "metadata": {
+        "id": "GGPF7cY3Ntyj"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Note:** The following cell may take about 2 minutes to run"
+      ],
+      "metadata": {
+        "id": "3tGnPucbzmEx"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#The following packages are used to run the example pipelines\n",
+        "from apache_beam.options.pipeline_options import PipelineOptions\n",
+        "\n",
+        "class MyDoFn(beam.DoFn):\n",
+        "  def process(self,element):\n",
+        "          returned_record = MyApiCall.get_data(\"http://my-api-call.com\")\n",
+        "          if len(returned_record)!=10:\n",
+        "            raise ValueError(\"Length of record does not match expected length\")\n",
+        "          yield returned_record\n",
+        "\n",
+        "with beam.Pipeline() as p3:\n",
+        "  result = (\n",
+        "          p3\n",
+        "          | ReadFromText(\"/content/sample_data/anscombe.json\")\n",
+        "          | beam.ParDo(MyDoFn())\n",
+        "          | WriteToText(\"/content/example1\")\n",
+        "  )"
+      ],
+      "metadata": {
+        "id": "Ktk9EVIFzGfP"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "**Mocking Example**\n",
+        "\n",
+        "The following blocks of code illustrate how we can mock an API response, to test out the error message we've written. Note that we can use mocking to avoid making the actual API call in our test."
```

Comment:
```suggestion
        "To test the error message, mock an API response, as demonstrated in the following blocks of code. Use mocking to avoid making the actual API call in the test."
```